src/admin/menu/views/MenuDetail.tsx

```diff
@@ -61,13 +61,13 @@ const MenuDetail: FunctionComponent<MenuDetailProps> = ({ history, match }) => {
 	const menuId = match.params.menu;
 
 	// Methods
-	const handleDelete = (refetch: () => void): void => {
+	const handleDelete = (refetchMenuItems: () => void): void => {
 		triggerMenuItemDelete({
 			variables: { id: idToDelete },
 			update: ApolloCacheManager.clearNavElementsCache,
 		})
 			.then(() => {
-				refetch();
+				setTimeout(refetchMenuItems); // Wait for cache to be fully cleared
```

Comment:
Wordt die refetch niet al in de DataQueryComponent gewrapped in een setTimeout?

Reply:
ik dacht toch nog ergens te zien dat een item niet direct weg was nadat je het verwijdert hebt. Maar ik zal het ff weg doen, dan kunnen we het grondiger onderzoeken als we de bug opmerken